src/Header.php

```diff
@@ -47,19 +47,21 @@ public static function parse($header): array
      */
     public static function normalize($header): array
     {
-        if (!is_array($header)) {
-            return array_map('trim', explode(',', $header));
-        }
-
         $result = [];
-        foreach ($header as $value) {
+        foreach ((array) $header as $value) {
```

Comment:
This first loop is a bit odd, because it appears to allow passing in an array of arrays. It is not useful to parse more than one type of header at once, as the list syntax is *not* defined for every header. i.e. simply passing in the full `->getHeaders()` array is not valid.

All in all this `normalize()` function is not documented super-well, the method name is not really helpful and I don't really trust the parsing logic based on regular expressions.

I suggest that this method is deprecated and replaced by a `public static function splitList(array|string $header)` that either accepts a `string` or a `string[]`, properly validating the input. The parser should be handwritten (simply handling quoted strings is easy), because that regular expression is pretty magic.

I can perform this change, but would like to hear your opinion first.

Reply:
I am reluctant to make big changes here. Can we go with whatever the minimum changes are to get your use case to work?

Reply:
The _current_ diff as it is in this PR contains the minimally necessary changes to _correctly_ implement parsing of headers containing lists.

My proposed change with the deprecation of the `normalize()` method would just be to tighten up the API and to clearly define the method's behavior.

Reply:
I just noticed your comment over in WoltLab/WCF (notifications for that repository go to $companymail):

This PR is not fixing an actual issue with our use case. This PR primarily is a matter of standard's correctness. The only real bug is when a string instead of an array is passed to `normalize()`, because then the simply `explode()` will be used.

I've also re-added the php-http Slack to my Slack Desktop Client. Feel free to message me there if you feel that it would help.